### DISCUSSION TITLE
Dual total energy counter

### DISCUSSION THREAD

**Discussion by lk29** [arendst/Tasmota#19629] [Q&A] [answered]
Perhaps someone has already described the method.  

How to display three total energy meters 

1 - 23:00:00-06:59:59 
2 - 07:00:00-22:59:59 
3 - the sum of the first and second

**Comment by wjohn007** [comment, marked as answer]
I had a similar requirement.
I was looking for a solution to display the hourly and daily consumption of electrical energy.
Tasmota provides the raw data. The graphical display should also be implemented under the Tasmota UI.

I defined a new Button "Historical data and statistics" in the main-menu.
![grafik](https://github.com/arendst/Tasmota/assets/115527085/1a691ed5-1ccb-4868-949f-5b339f6c0263)

Tasmota forwards the raw values ​​to [thingspeak](https://thingspeak.com/).

Thingspeak saves the data, aggregates it as desired  and generates a graphic that Tasmota can integrate via iframe. (scripting is required) 

This gives you the following representation, for example:
![grafik](https://github.com/arendst/Tasmota/assets/115527085/3f945663-6e24-40da-9ea1-a625f9b6f73a)

I'm pretty sure that this will also meet your requirements.

John

**Reply by lk29** [reply]
beautiful but redundant and requires third-party service.

For this I can use mqtt or homeassistant. 

I’m looking for a simple solution to take into account the total consumption but divided by time (night and day tariff)

**Comment by wjohn007** [comment]
For tasmota-only solutions berry-script my be helpful.
John

**Reply by lk29** [reply]
do you have examples?

**Reply by lk29** [reply]
> For tasmota-only solutions berry-script my be helpful.
> John

Use esp8266ex 
~~Berry scripting not supported:(~~

Make custom! 
I'm trying

**Reply by sfromis** [reply]
More generally, a reason why Tasmota does not facilitate doing what you'd like to do is that the deliberately rudimentary web UI is not intended to be used as a "device dashboard" for daily operation, which is instead expected to be handled in your backend platform with much better resources available.

**Reply by lk29** [reply]
But if there is a “web interface”, then why not use it to the fullest?  In my project I would like to maintain minimal dependence on third-party software.

**Reply by sfromis** [reply]
Because Tasmota was designed for ESP8266 with **very limited** resources, the priority was ability to handle controls and sensors, not a flexible web UI. There are reasons why the "M" in the "Tasmota" name means "MQTT" :wink: 

To be clear, I'm not trying to say what *you* "should or should not", merely explaining the principles not including being a standalone server with rich features for growing a "device dashboard".

**Comment by tiagofreire-pt** [comment]
Did anyone get a local berry script solution?

**Reply by s-hadinger** [reply]
I will develop a generic berry energy driver so you can do everything in berry in a clean and supported way. But not before end of August

**Reply by tiagofreire-pt** [reply]
Would you consider this? https://github.com/arendst/Tasmota/discussions/21922
